File: src/main/java/de/bettinggame/adapter/AbstractController.java
package de.bettinggame.adapter;

import org.springframework.web.bind.annotation.ModelAttribute;

import de.bettinggame.ui.Navigation;

import java.util.List;

/**
 * Abstract base controller.
 */
public interface AbstractController {

    /**
     * Returns navigation items
     */
    @ModelAttribute("nonRestrictedNavigation")
    default List<Navigation> getNonRestrictedNavigation() {
        return Navigation.Companion.getNonRestrictedNavigation();
    }

    @ModelAttribute("userRestrictedNavigation")
    default List<Navigation> getUserRestrictedNavigation() {
        return Navigation.Companion.getUserRestrictedNavigation();
    }

    @ModelAttribute("adminRestrictedNavigation")
    default List<Navigation> getAdminRestrictedNavigation() {
        return Navigation.Companion.getAdminRestrictedNavigation();
    }
}

File: src/main/kotlin/de/bettinggame/adapter/AbstractController.kt
package de.bettinggame.adapter

import de.bettinggame.ui.Navigation
import org.springframework.web.bind.annotation.ModelAttribute

interface AbstractController {
    /**
     * Returns navigation items
     */
    @JvmDefault
    @ModelAttribute("nonRestrictedNavigation")
    fun getNonRestrictedNavigation(): List<Navigation> = Navigation.getNonRestrictedNavigation()

    @JvmDefault
    @ModelAttribute("userRestrictedNavigation")
    fun getUserRestrictedNavigation(): List<Navigation> = Navigation.getUserRestrictedNavigation()

    @JvmDefault
    @ModelAttribute("adminRestrictedNavigation")
    fun getAdminRestrictedNavigation(): List<Navigation> = Navigation.getAdminRestrictedNavigation()
}
